src/main.js

```diff
@@ -1,241 +1,56 @@
-        </header>
-
-        <nav class="bg-white shadow-md p-2 sticky top-14">
-          <ul class="flex justify-around">
-            <li><a href="/" class="text-gray-600">홈</a></li>
-            <li><a href="/profile" class="text-blue-600">프로필</a></li>
-            <li><a href="#" class="text-gray-600">로그아웃</a></li>
-          </ul>
-        </nav>
-
-        <main class="p-4">
-          <div class="bg-white p-8 rounded-lg shadow-md">
-            <h2 class="text-2xl font-bold text-center text-blue-600 mb-8">
-              내 프로필
-            </h2>
-            <form>
-              <div class="mb-4">
-                <label
-                  for="username"
-                  class="block text-gray-700 text-sm font-bold mb-2"
-                  >사용자 이름</label
-                >
-                <input
-                  type="text"
-                  id="username"
-                  name="username"
-                  value="홍길동"
-                  class="w-full p-2 border rounded"
-                />
-              </div>
-              <div class="mb-4">
-                <label
-                  for="email"
-                  class="block text-gray-700 text-sm font-bold mb-2"
-                  >이메일</label
-                >
-                <input
-                  type="email"
-                  id="email"
-                  name="email"
-                  value="hong@example.com"
-                  class="w-full p-2 border rounded"
-                />
-              </div>
-              <div class="mb-6">
-                <label
-                  for="bio"
-                  class="block text-gray-700 text-sm font-bold mb-2"
-                  >자기소개</label
-                >
-                <textarea
-                  id="bio"
-                  name="bio"
-                  rows="4"
-                  class="w-full p-2 border rounded"
-                >
-안녕하세요, 항해플러스에서 열심히 공부하고 있는 홍길동입니다.</textarea
-                >
-              </div>
-              <button
-                type="submit"
-                class="w-full bg-blue-600 text-white p-2 rounded font-bold"
-              >
-                프로필 업데이트
-              </button>
-            </form>
-          </div>
-        </main>
-
-        <footer class="bg-gray-200 p-4 text-center">
-          <p>&copy; 2024 항해플러스. All rights reserved.</p>
-        </footer>
-      </div>
-    </div>
-  </div>
-`;
-
-document.body.innerHTML = `
-  ${MainPage()}
-  ${ProfilePage()}
-  ${LoginPage()}
-  ${ErrorPage()}
-`;
+import { $ } from "./utils/dom";
+import { user } from "./store";
+import { ROUTES_MAP } from "./router/routes";
+import { BrowserRouter } from "./router/router";
+
+const router = new BrowserRouter({
+  guardRoute: (route, next) => {
+    if (route.path === ROUTES_MAP.LOGIN && user.loggedIn()) {
+      next(ROUTES_MAP.HOME);
+    } else if (route.authRequired && !user.loggedIn()) {
+      next(ROUTES_MAP.LOGIN);
+    } else {
+      next();
+    }
```

Comment:
조건절 이후에 별도 로직이 없다면 
if + return 을 사용해서 가독성을 높힐수 있다는 생각입니다!

```suggestion
    if (route.path === ROUTES_MAP.LOGIN && user.loggedIn()) {
      next(ROUTES_MAP.HOME);
      
      return;
    } 
    
    if (route.authRequired && !user.loggedIn()) {
      next(ROUTES_MAP.LOGIN);
      
      return;
    } 
    
    next();
```

Reply:
저도 early return 좋아합니다!